lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -0,0 +1,300 @@
+        }
+    }
+
+    /**
+     * Function which validates server authorization token response.
+     * @param serverResponse
+     */
+    validateServerAuthorizationTokenResponse(
+        serverResponse: ServerAuthorizationTokenResponse
+    ): void {
+        // Check for error
+        if (serverResponse.error || serverResponse.error_description) {
+            const errString = `${serverResponse.error_codes} - [${serverResponse.timestamp}]: ${serverResponse.error_description} - Correlation ID: ${serverResponse.correlation_id} - Trace ID: ${serverResponse.trace_id}`;
+            throw new ServerError(serverResponse.error, errString);
+        }
+
+        // generate homeAccountId
+        if (serverResponse.client_info) {
+            this.clientInfo = buildClientInfo(
+                serverResponse.client_info,
+                this.cryptoObj
+            );
+            if (
+                !StringUtils.isEmpty(this.clientInfo.uid) &&
+                !StringUtils.isEmpty(this.clientInfo.utid)
+            ) {
+                this.homeAccountIdentifier =
+                    this.cryptoObj.base64Encode(this.clientInfo.uid) +
+                    "." +
+                    this.cryptoObj.base64Encode(this.clientInfo.utid);
+            }
+        }
+    }
+
+    /**
+     * Returns a constructed token response based on given string. Also manages the cache updates and cleanups.
+     * @param serverTokenResponse
+     * @param authorityString
+     * @param resource
+     * @param state
+     */
+    generateAuthenticationResult(
+        serverTokenResponse: ServerAuthorizationTokenResponse,
+        authority: Authority
+    ): AuthenticationResult {
+        // Retrieve current account if in Cache
+        // TODO: add this once the req for cache look up for tokens is confirmed
+
+        const authenticationResult = this.processTokenResponse(
+            serverTokenResponse,
+            authority
+        );
+
+        const environment =
+            authority.canonicalAuthorityUrlComponents.HostNameAndPort;
+        this.addCredentialsToCache(authenticationResult, environment);
+
+        return authenticationResult;
+    }
+
+    /**
+     * Returns a new AuthenticationResult with the data from original result filled with the relevant data.
+     * @param authenticationResult
+     * @param idTokenString(raw idToken in the server response)
+     */
+    processTokenResponse(
+        serverTokenResponse: ServerAuthorizationTokenResponse,
+        authority: Authority
+    ): AuthenticationResult {
+        const authenticationResult: AuthenticationResult = {
+            uniqueId: "",
+            tenantId: "",
+            tokenType: "",
+            idToken: null,
+            idTokenClaims: null,
+            accessToken: "",
+            refreshToken: "",
+            scopes: [],
+            expiresOn: null,
+            familyId: null,
+        };
+
+        // IdToken
```

Comment:
nit: Unnecessary comment here (and a couple of other places)

Reply:
deleted some. Kept some. Lets rehaul comments later.